### PR TITLE
avatar: Make avatars more of a utility (with settings)

### PR DESCRIPTION
**Pull request by greezybacon**
Initially, include support only for Gravatar. This will also allow for more flexible avatar sources, including LDAP and Active Directory.

### Outstanding
* [x] Fix avatar sizing on the client portal

### CODE REVIEW

File: include/class.avatar.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+**********************************************************************/
+
+abstract class Avatar {
+    var $user;
+
+    function __construct($user) {
+        $this->user = $user;
+    }
+
+    abstract function getUrl($size);
```
Comment (greezybacon) [inline include/class.avatar.php]: The interface should add `__toString()` since the system coerces them to strings

**Comment by Chefkeks** [comment]
Seems like Avatars are growing up :+1: :)

**Comment by jdelhome3578** [comment]
Ticket Thread Client View Error

image

File: include/class.avatar.php [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+    function __construct($mode) {
+        $this->mode = $mode;
+    }
+
+    function makeAvatar($uid) {
+        if (!extension_loaded('gd'))
+            Http::redirect(ROOT_PATH.'images/mystery-oscar.png');
+
+        $sprite = self::$sprites[$this->mode];
```
Comment (greezybacon) [inline include/class.avatar.php]: Show mystery oscar for unknown mode

**Comment by jdelhome3578** [comment]
Ticket Thread Client View after message post.

image

**Pull request merged**